nemo/core/classes/common.py

```diff
@@ -74,11 +75,39 @@ def _attach_and_validate_output_types(self, out_objects):
         # TODO: Properly implement this
```

Comment:
To double-check that I understand this function, would the following docstring make sense?
`
This function does two things. 1) It ensures that len(out_object) == len(self.output_types). 2) If the output is a tensor (or list/tuple of list/tuple ... of tensors), it attaches a neural_type to it. Note: tensor.neural_type is only checked on during _validate_input_types which is called prior to forward().
`

Reply:
This is a good start. Will add it with some clarifications. Technically we can attach types to anything which is an object of a class, not just tensors.

Reply:
Done